presto-main/src/main/java/com/facebook/presto/sql/gen/ByteCodeUtils.java

```diff
@@ -244,13 +247,14 @@ else if (type == Void.class) {
 
     public static ByteCodeNode invoke(CompilerContext context, Binding binding, String name)
     {
+        checkIdentifier(name);
         return new Block(context)
                 .invokeDynamic(name, binding.getType(), binding.getBindingId());
     }
 
     public static ByteCodeNode invoke(CompilerContext context, Binding binding, Signature signature)
     {
-        return invoke(context, binding, signature.getName());
+        return invoke(context, binding, mungeIdentifier(signature.getName()));
```

Comment:
This the method that takes a Presto `Signature` which can contain invalid Java names. The other method that takes a name only does verification.